File: backend/src/main/java/com/project/newsfilter/dto/NewsDTO.java
package com.project.newsfilter.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class NewsDTO {

    // News source name (e.g. CNN, Fox News)
    private String name;
    // Title of article
    private String title;
    // Description of article
    private String description;
    // URL of article
    private String url;
    // URL of image
    private String urlToImage;
    //  Shortened article content
    private String content;
    // date published
    private String publishedAt;
}

File: frontend/newsfilter-ui/src/app/news-dto.ts
export interface NewsDto{

  // News source name (e.g. CNN, Fox News)
  name: string;
  // Title of article
  title: string;
  // Description of article
 description: string;
  // URL of article
  url: string;
  // URL of image
  urlToImage: string;
  //  Shortened article content
  content: string;
  // date published
  publishedAt: string;
}
